File: FleetService/Models/Events/OrderNotificationEvent.cs
namespace FleetService.Models.Events;

public class OrderNotificationEvent
{
    public Guid Id { get; set; }
    public string Microservice { get; set; } = "order-ms";
    public string Action { get; set; }
    public Guid OrderId { get; set; }
    public string EntityType { get; set; } = "ORDER";
    public string Message { get; set; }
    public DateTime Timestamp { get; set; }
    public Dictionary<string, object> Data { get; set; } = new();
    public string Severity { get; set; }
    
    public string GetTimeStamp()
    {
        return Timestamp.ToString("yyyy-MM-ddTHH:mm:ss");
    }
}

File: order-ms/src/main/java/com/espe/edu/ec/order_ms/dtos/events/OrderNotificationEvent.java
package com.espe.edu.ec.order_ms.dtos.events;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.Map;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class OrderNotificationEvent {

    private UUID id;
    private final String microservice = "order-ms";
    private String action;

    private UUID orderId;
    private final String entityType = "ORDER";
    
    private String message;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime timestamp;

    private Map<String, Object> data;
    private String severity;

    public String getTimeStamp(){
        return timestamp != null ? 
        timestamp.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME) : 
        LocalDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE_TIME);
    }



}

